src/main/java/tech/units/indriya/function/DefaultNumberSystem.java

```diff
@@ -460,13 +460,16 @@ public Number narrow(Number number) {
 
         if(number instanceof BigDecimal) {
             
-            final BigDecimal decimal = ((BigDecimal) number);
-            try {
-                BigInteger integer = decimal.toBigIntegerExact(); 
+            BigDecimal decimal = ((BigDecimal) number);
+            // see https://stackoverflow.com/questions/1078953/check-if-bigdecimal-is-integer-value
+            decimal = decimal.stripTrailingZeros();
```

Comment:
Note: `stripTrailingZeros` creates a new `BigDecimal` just to do the check. So it improves performance when given `BigDecimal` cannot be converted to `BigInteger`. However it worsens performance otherwise.

Reply:
... but that's ok as in most cases when this code block is executed, the BigDecimal will have fractional parts anyway.